sw/host/opentitanlib/src/transport/cw310/mod.rs

```diff
@@ -231,6 +231,21 @@ impl Transport for CW310 {
             usb.pll_out_enable(2, false)?;
             usb.pll_write_defaults()?;
             Ok(None)
+        } else if let Some(reset) = action.downcast_ref::<Reset>() {
+            // Note: `TransportWrapper::apply_pin_strapping` would be better but this is in
+            // `Transport`.
+            log::info!("Asserting the reset signal");
+            let reset_pin = self.gpio_pin(Self::PIN_SRST)?;
+            reset_pin.write(false)?;
+            std::thread::sleep(reset.reset_delay);
+            if reset.clear_uart_rx {
+                log::info!("Clearing the UART RX buffer");
+                self.uart("0")?.clear_rx_buffer()?;
+            }
+            log::info!("Deasserting the reset signal");
+            reset_pin.write(true)?;
+            std::thread::sleep(reset.reset_delay);
+            Ok(None)
```

Comment:
Its unclear to me why this is a `dispatch` operation.  Since the `clear_rx_buffer` is a feature of the `Uart` trait, wouldn't this be better implemented generically in `TransportWrapper` something like:

```
impl TransportWrapper {
  fn reset_target(&self, reset_delay: Duration, clear_uart: bool) {
    apply_pin_strapping(...)
    sleep(...)
    if clear_uart { ... }
    (etc)
  }
}
```

Reply:
I added this as a dispatch operation since I wasn't sure how it would work with transports like proxy or ti50emulator. @cfrantz explained me offline that they have GPIO simulation and we can also handle special cases by downcasting to the specific transport type.